evalml/pipelines/time_series_classification_pipelines.py

```diff
@@ -146,13 +166,8 @@ def score(self, X, y, objectives):
         y_encoded = self._encode_targets(y)
         y_shifted = y_encoded.shift(-self.gap)
         y_pred, y_pred_proba = self._compute_predictions(X, y, objectives)
-        non_nan_mask = _get_rows_without_nans(y_shifted, y_pred, y_pred_proba)
-        if y_pred is not None:
-            y_pred = y_pred.iloc[non_nan_mask]
-        if y_pred_proba is not None:
-            y_pred_proba = y_pred_proba.iloc[non_nan_mask]
-        y_labels = y_shifted.iloc[non_nan_mask]
```

Comment:
Is it necessary to remove the `iloc` stuff from where this code got moved to?

Reply:
The calls to `iloc` are now happening in `drop_rows_with_nans`!